### DISCUSSION TITLE
Current implementation pattern for ExportMetrics

### DISCUSSION THREAD

**Discussion by andrewang-canva** [open-telemetry/opentelemetry-java#6341] [Q&A]
Hi there,

I want to export metrics emitted by [ExportMetrics](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/common/src/main/java/io/opentelemetry/exporter/internal/ExporterMetrics.java)

Wondering what is current approved implementation for _ExportMetrics_ via OpenTelemetrySdk? Currently we are leveraging on the sdk via the following:

```
var exporter = OtlpGrpcMetricExporter.builder()
        .setEndpoint("http://localhost:4317")
        .build();

var reader = PeriodicMetricReader.builder(exporter)
    .setInterval(Duration.ofSeconds(10)).build();

SdkMeterProviderUtil.registerMetricReaderWithCardinalitySelector(
    sdkMeterProviderBuilder, reader, unused -> 100_000);

sdkMeterProviderBuilder.registerView(
    InstrumentSelector.builder().setMeterName("ExponentialHistogram").build(),
      View.builder().setAggregation(Aggregation.base2ExponentialBucketHistogram()).build());

var sdkMeterProvider = sdkMeterProviderBuilder.build();

openTelemetrySdkClient = OpenTelemetrySdk.builder()
    .setMeterProvider(sdkMeterProvider)
    .build();
```

Cheers

**Comment by jack-berg** [comment]
That looks about right to me. 

A few notes about `ExporterMetrics`:
- The metrics produced by this are not stable and will see breaking changes at some point in the future.
- ExporterMetrics is used for `OtlpHttpSpanExporter`, `OtlpGrpcSpanExporter`, `OtlpHttpLogRecordExporter`, `OtlpGrpcLogRecordExporter`, and `ZipkinSpanExproter`. It _is not_ used for `OtlpHttpMetricExporter` or `OtlpGrpcMetricExporter` - there are currently no metrics for tracking metric export.
- ExporterMetrics will be enabled by default if you use `opentelemetry-sdk-extension-autoconfigure`, but if you're programmatically configuring the SDK like in your example, you need to make sure you register your SdkMeterProvider with your exporters. For example, to enable ExporterMetrics for `OtlpHttpSpanExporter`, call `OtlpHttpSpanExporterBuilder#setMeterProvider(sdkMeterProvider)`. This requires careful ordering during initialization, since SdkMeterProvider needs to be initialized before SdkLoggerProvider / SdkTracerProvider.

**Reply by andrewang-canva** [reply]
Thanks @jack-berg, I was not aware that `ExportMetrics` is not available for tracking metric export which is what i'm looking for in my application. If that's the case, are there any other internal metrics that we can leverage on?
